### PR TITLE
set golang to 1.22.5

### PR DESCRIPTION
**Pull request by pfrcks**
This pull request includes updates to the Go version used in various parts of the project. The Go version has been updated from `1.19.x` or `1.20.14` to `1.22.5` in different files. 

Go version updates:

* [`.github/workflows/run_unit_tests.yml`](diffhunk://#diff-394fcfd605d72aeb92748c54045208916c0dd638a4c1e09f2b15cc4ceffbcc48L16-R19): Updated the Go version used in the `Golang-Tests` job from `1.19.x` to `1.22.5`.
* [`.pipelines/azure_pipeline_mergedbranches.yaml`](diffhunk://#diff-2ee1d450a2b4018224d6c91140a21692f8e2fe7652f9c06359493a9fe8a2785dR278-R282): Added a task in the jobs to find in cache or download the updated Go version `1.22.5` and add it to the PATH.
* [`scripts/build/windows/install-build-pre-requisites.ps1`](diffhunk://#diff-3b93f158c0288f1eaaea49c9f5d439061deee1ecf863a4a7bb3041ee97f313b8L17-R17): Updated the Go version in the `Install-Go` function and the corresponding message in the `Install-cmetrics` function from `1.20.14` to `1.22.5`. [[1]](diffhunk://#diff-3b93f158c0288f1eaaea49c9f5d439061deee1ecf863a4a7bb3041ee97f313b8L17-R17) [[2]](diffhunk://#diff-3b93f158c0288f1eaaea49c9f5d439061deee1ecf863a4a7bb3041ee97f313b8L158-R158)

### CODE REVIEW

**Comment by pfrcks** [comment]
validated no perf impact in both legacy and msi modes. also validated fluentbit input plugins work well

**Review by wanlonghenry — DISMISSED** [review]

File: source/plugins/go/input/go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module Docker-Provider/source/plugins/go/input
 
-go 1.19
+go 1.21.0
```
Comment (ganga1980) [inline source/plugins/go/input/go.mod:3]: what's the reason for using 1.21.0 version here?

Comment (pfrcks) [reply source/plugins/go/input/go.mod:3]: this change is made by go mod tidy when using go version 1.22.x

File: source/plugins/go/src/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module Docker-Provider/source/plugins/go/src
 
-go 1.19
+go 1.21
```
Comment (ganga1980) [inline source/plugins/go/src/go.mod:3]: same here, what's reason for using go version 1.21 instead of 1.22.5?

Comment (pfrcks) [reply source/plugins/go/src/go.mod:3]: replied above

**Review by ganga1980 — DISMISSED** [review]

**Review by ganga1980 — APPROVED** [review]

**Pull request merged**